src/debug_wait.c

```diff
@@ -108,6 +108,23 @@ ts_debug_waitpoint_wait(DebugWait *waitpoint, bool blocking)
 	ereport(DEBUG3, (errmsg("proceeding after waitpoint '%s'", waitpoint->tagname)));
 }
 
+void
+ts_debug_waitpoint_error_if_enabled(DebugWait *waitpoint)
+{
+	LockAcquireResult lock_acquire_result = LockAcquire(&waitpoint->tag, ExclusiveLock, true, true);
+
+	if (lock_acquire_result == LOCKACQUIRE_NOT_AVAIL)
+		return;
```

Comment:
Shouldn't it error right here?  `LOCKACQUIRE_NOT_AVAIL` would mean that this waitpoint was enabled and `ExclusiveLock` should conflict with itself and return `LOCKACQUIRE_NOT_AVAIL`. It shouldn't ever return `LOCKACQUIRE_ALREADY_HELD` in this scenario as far as I understand.

Reply:
I think it a good idea to test it using isolation tests as well, using separate sessions